.github/workflows/python-publish.yml

```diff
@@ -5,7 +5,95 @@ on:
     tags:
       - "v*"
 
+permissions:
+  contents: read
+
+env:
+  python_interpreters: 3.7 3.8 3.9 3.10 3.11
```

Comment:
can we start testing 3.12?

Reply:
Let's do this in a separate PR and enable it for the core package as well.